frontend_tests/node_tests/popover_menus_data.js

```diff
@@ -113,140 +148,322 @@ function test(label, f) {
+        move_message_menu_item: undefined,
+        should_display_mark_as_unread: true,
+        view_source_menu_item: undefined,
+        should_display_collapse: true,
+        should_display_uncollapse: false,
+        should_display_add_reaction_option: true,
+        should_display_edit_history_option: true,
+        should_display_hide_option: false,
+        conversation_time_uri: "http://zulip.zulipdev.com/undefined/near/1",
+        narrowed: false,
+        should_display_delete_option: true,
+        should_display_read_receipts_option: true,
+        should_display_reminder_option: false,
+        should_display_quote_and_reply: true,
+    });
 
-    const response = popover_menus_data.get_actions_popover_content_context(1);
-    assert.equal(response.message_id, 1);
-    assert.equal(response.stream_id, 1);
-    assert.equal(response.editability_menu_item, "translated: Edit message");
-    assert.equal(response.move_message_menu_item, "translated: Move messages");
-    assert.equal(response.should_display_mark_as_unread, true);
-    assert.equal(response.view_source_menu_item, undefined);
-    assert.equal(response.should_display_collapse, true);
-    assert.equal(response.should_display_uncollapse, false);
-    assert.equal(response.should_display_add_reaction_option, true);
-    assert.equal(response.should_display_edit_history_option, true);
-    assert.equal(response.should_display_hide_option, false);
-    assert.equal(response.conversation_time_uri, "http://zulip.zulipdev.com/undefined/near/1");
-    assert.equal(response.narrowed, false);
-    assert.equal(response.should_display_delete_option, true);
-    assert.equal(response.should_display_read_receipts_option, true);
-    assert.equal(response.should_display_reminder_option, false);
-    assert.equal(response.should_display_quote_and_reply, true);
-});
-
-test("not_my_message_view_actions", () => {
-    set_page_params_no_edit_restrictions();
-    // Get message that is only viewable
-
-    const list = init_message_list();
-    message_lists.set_current(list);
-
-    // Message is sent by somebody else and is a stream with previous history.
-    // I should only be able to view this message with no edit/move permissions.
-    const messages = [
-        {
-            id: 1,
-            sender_id: mike.user_id,
-            is_hidden: false,
-            sent_by_me: false,
-            locally_echoed: false,
-            is_stream: true,
-            stream_id: 1,
-            collapsed: false,
-            unread: false,
-            edit_history: [
-                {
-                    prev_content: "Previous content",
-                    prev_stream: 0,
-                    prev_topic: "Previous topic",
-                },
-            ],
-        },
-    ];
+    reset_page_params_user_status();
+    page_params.is_guest = true;
+    response = popover_menus_data.get_actions_popover_content_context(3);
+    assert.deepEqual(response, {
```

Comment:
Let's add a comment before each test to describe which type of user we are testing.